packages/website/components/mdx/tabs.js

```diff
@@ -0,0 +1,133 @@
+import React, { useState, cloneElement, Children, isValidElement } from 'react'
+
+import clsx from 'clsx'
+
+/**
+ * @typedef {object} TabItemProps
+ * @property {React.ReactNode} children
+ * @property {string} value a unique id for this tab item
+ * @property {boolean} [default] if true, sets this item as the default on first load
+ * @property {string} [label] a label to display in the tab header instead of props.value
+ * @property {boolean} [hidden]
+ * @property {string} [className]
+ * @property {Record<string, unknown>} [attributes]
+ */
+
+/**
+ *
+ * @param {TabItemProps} props
+ */
+export function TabItem(props) {
+  const { hidden, className, children } = props
+  return (
+    <div role="tabpanel" {...{ hidden, className: clsx(className) }}>
+      {children}
+    </div>
+  )
+}
+
+/**
+ * @typedef {object} TabsProps
+ * @property {React.ReactNode} children
+ * @property {string} [className]
+ */
+
+/**
+ *
+ * @param {TabsProps} props
+ */
+export function Tabs(props) {
+  const children =
+    Children.map(props.children, (child) => {
+      if (isValidElement(child) && isTabItem(child)) {
+        return child
+      }
+      throw new Error(
+        `Unexpected child in Tabs element. All children of <Tabs> should be a <TabItem> with a unique "value" prop`
+      )
+    }) || []
+
+  const values = children.map(({ props: { value, label, attributes } }) => ({
```

Comment:
suggestion (non-blocking): Some of this might be good to memoize, I haven't run this in a profiler, but small calculation functions usually work well with memoization for minor performance improvements.